core/ledger/pvtdatastorage/store.go

```diff
@@ -801,22 +810,31 @@ func (s *Store) retrieveExpiryEntries(minBlkNum, maxBlkNum uint64) ([]*expiryEnt
 
 func (s *Store) launchCollElgProc() {
 	go func() {
-		s.processCollElgEvents() // process collection eligibility events when store is opened - in case there is an unprocessed events from previous run
+		if err := s.processCollElgEvents(); err != nil {
+			// process collection eligibility events when store is opened -
+			// in case there is an unprocessed events from previous run
+			logger.Errorw("failed to process collection eligibility events", "err", err)
+		}
```

Comment:
Thanks @cendhu. It will be good to add a test for this log event, as this is kind of part of the behavior that we count on. I am fine merging this PR, has it has wide spread change. You can do this in a separate PR.

Reply:
Done.